x-pack/plugins/infra/public/pages/metrics/hosts/hooks/use_hosts_table.tsx

```diff
@@ -140,7 +140,7 @@ export const useHostsTable = () => {
   } = useKibanaContextForPlugin();
   const { dataView } = useMetricsDataViewContext();
 
-  const [hostFlyoutState, setHostFlyoutState] = useHostFlyoutUrlState();
+  const [hostFlyoutState, setHostFlyoutState] = useAssetDetailsUrlState();
```

Comment:
We're only using the `useAssetDetailsUrlState` to get the `itemId` and control the flyout's visibility. It seems like this attribute should be owned by the `use_hosts_table`. 

The URL state I'm thinking of is the `useHostsTableUrlState`. Looks like `itemId` could be moved to that hook .